adapters/repos/db/roaringsetrange/segment_reader.go

```diff
@@ -23,14 +23,23 @@ import (
 )
 
 type SegmentReader struct {
-	cursor   SegmentCursor
-	sroarBuf []uint16
+	cursor    SegmentCursor
+	sroarBufs [][]uint16
 }
 
 func NewSegmentReader(cursor *GaplessSegmentCursor) *SegmentReader {
+	return NewSegmentReaderConcurrent(cursor, 1)
+}
+
+func NewSegmentReaderConcurrent(cursor *GaplessSegmentCursor, concurrency int) *SegmentReader {
+	sroarBufs := make([][]uint16, concurrency)
+	for i := range sroarBufs {
+		sroarBufs[i] = make([]uint16, 4100) // sroar.maxContainerSize
```

Comment:
4100 == sroar.maxContainerSize, yes?

Reply:
correct. hardcoded, cause `maxContainerSize` is not exported